Cryptography upstream recently switched the build system to Maturin,
which is not available on Solaris and is not fully compatible with
it at the moment.

Since its integration will take some effort and we don't want to delay
this update any longer, the patch reverts the switch.

https://github.com/pyca/cryptography/commit/5b23baae7e1a59ab16630e6f8eae13e2b8822ded

--- cryptography-43.0.1/pyproject.toml
+++ cryptography-43.0.1/pyproject.toml
@@ -2,15 +2,14 @@
 # These requirements must be kept sync with the requirements in
 # ./github/requirements/build-requirements.{in,txt}
 requires = [
-    "maturin>=1,<2",
+    # First version of setuptools to support pyproject.toml configuration
+    "setuptools>=61.0.0",
 
     # Must be kept in sync with `project.dependencies`
     "cffi>=1.12; platform_python_implementation != 'PyPy'",
-    # Needed because cffi imports distutils, and in Python 3.12, distutils has
-    # been removed from the stdlib, but installing setuptools puts it back.
-    "setuptools!=74.0.0,!=74.1.0,!=74.1.1",
+    "setuptools-rust>=1.7.0",
 ]
-build-backend = "maturin"
+build-backend = "setuptools.build_meta"
 
 [project]
 name = "cryptography"
@@ -58,6 +57,14 @@ source = "https://github.com/pyca/crypto
 issues = "https://github.com/pyca/cryptography/issues"
 changelog = "https://cryptography.io/en/latest/changelog/"
 
+[tool.setuptools]
+zip-safe = false
+package-dir = {"" = "src"}
+
+[tool.setuptools.packages.find]
+where = ["src"]
+include = ["cryptography*"]
+
 [project.optional-dependencies]
 ssh = ["bcrypt >=3.1.5"]
 
@@ -79,42 +86,11 @@ sdist = ["build"]
 # `click` included because its needed to type check `release.py`
 pep8test = ["ruff", "mypy", "check-sdist", "click"]
 
-[tool.maturin]
-python-source = "src"
-python-packages = ["cryptography"]
-manifest-path = "src/rust/Cargo.toml"
-module-name = "cryptography.hazmat.bindings._rust"
-locked = true
-sdist-generator = "git"
-features = ["pyo3/abi3-py37"]
-include = [
-    "CHANGELOG.rst",
-    "CONTRIBUTING.rst",
-    "LICENSE",
-    "LICENSE.APACHE",
-    "LICENSE.BSD",
-
-    "docs/**/*",
-
-    "src/_cffi_src/**/*.py",
-    "src/_cffi_src/**/*.c",
-    "src/_cffi_src/**/*.h",
-
-    "src/rust/**/Cargo.toml",
-    "src/rust/**/Cargo.lock",
-    "src/rust/**/*.rs",
-
-    "tests/**/*.py",
-]
-exclude = [
-    "vectors/**/*",
-    "src/rust/target/**/*",
-    "docs/_build/**/*",
-    ".github/**/*",
-    ".readthedocs.yml",
-    "ci-constraints-requirements.txt",
-    "mypy.ini",
-]
+[[tool.setuptools-rust.ext-modules]]
+target = "cryptography.hazmat.bindings._rust"
+path = "src/rust/Cargo.toml"
+py-limited-api = "auto"
+rust-version = ">=1.65.0"
 
 [tool.pytest.ini_options]
 addopts = "-r s --capture=no --strict-markers --benchmark-disable"
--- cryptography-43.0.1/MANIFEST.in
+++ cryptography-43.0.1/MANIFEST.in
@@ -0,0 +1,24 @@
+include CHANGELOG.rst
+include CONTRIBUTING.rst
+include LICENSE
+include LICENSE.APACHE
+include LICENSE.BSD
+include README.rst
+include noxfile.py
+
+include pyproject.toml
+recursive-include src py.typed *.pyi
+
+recursive-include docs *
+recursive-include src/_cffi_src *.py *.c *.h
+recursive-include src/rust Cargo.toml Cargo.lock *.rs
+prune docs/_build
+recursive-include tests *.py
+exclude vectors
+recursive-exclude vectors *
+exclude src/rust/target
+recursive-exclude src/rust/target *
+
+recursive-exclude .github *
+
+exclude release.py .readthedocs.yml ci-constraints-requirements.txt mypy.ini
